src/controllers/shared/common.resume.submission.controller.ts

```diff
@@ -104,12 +104,11 @@ const setWebApplicationData = (session: Session, appData: ApplicationData, trans
   appData[Transactionkey] = transactionId;
   appData[OverseasEntityKey] = overseaEntityId;
 
-  if (Object.keys(appData[OverseasEntityDueDiligenceKey] as OverseasEntityDueDiligence).length) {
+  if (appData[OverseasEntityDueDiligenceKey] && Object.keys(appData[OverseasEntityDueDiligenceKey]).length) {
```

Comment:
This part of the code comes from Mark, he is working on it right now.
It has been added to prevent the tests from failing, it has no impact on my fix.